src/compile/data.ts

```diff
@@ -564,7 +667,16 @@ export namespace summary {
     return [];
   }
 
-  export function assemble(component: DataComponent): VgData[] {
+  export function parseLayer(model: LayerModel): SummaryComponent[] {
+    // we create new data sources for each summary in a layer
+    return [];
```

Comment:
We should do it now, so that we don't need rename in assemble as commented below.